utils/timeSlot.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Utility file to handle time slots for scheduled appointments.
+ */
+
+import { TimeSlot } from '../types/common'
+
+/**
+ * Parse a time slot from the API gateway.
+ */
+export function parseTimeSlot(timeSlot: string): TimeSlot | null {
+  // Time slots are expected to be in the format 10-12.
+  const match = /(\d+)[-–—](\d+)/.exec(timeSlot)
```

Comment:
Note: This line doesn't render very well in Github. The `[-–—]` is saying that either a hyphen (`-`) or an ndash (`–`) or an mdash (`—`) are acceptable.